gateway/ln-gateway/src/lib.rs

```diff
@@ -1610,6 +1619,7 @@ impl IntoResponse for GatewayError {
                 "The gateway is disconnected from the Lightning Node".to_string(),
                 StatusCode::NOT_FOUND,
             ),
+            GatewayError::UnexpectedState(msg) => (msg, StatusCode::EXPECTATION_FAILED),
```

Comment:
Can we make a new `GatewayError` instead of re-using `UnexpectedState`? I think `UnexpectedState` should still be a 500 error code, since that was originally made to indicate that a state machine did not execute correctly.

Reply:
I made `GatewayError::NonZeroBalanceLeaveError` but still used `StatusCode::EXPECTATION_FAILED`, not sure if there's a better status code to use for this

Reply:
Works for me!